Makefile

```diff
@@ -12,13 +12,14 @@
 # See the License for the specific language governing permissions and
 
 GO := godep go
+goversion = 1.6.2
 pkgs  = $(shell $(GO) list ./...)
 
 all: format build test
 
 test:
-	@echo ">> running tests"
-	@$(GO) test -tags test -short -race $(pkgs)
+	@echo ">> running tests using docker"
+	@docker run -i -v $(PWD):/go/src/github.com/google/cadvisor golang:$(goversion) /bin/sh -c "cd /go/src/github.com/google/cadvisor && go get github.com/tools/godep && $(GO) test -tags test -short -race $(pkgs)"
```

Comment:
Perhaps use `-w` (workdir) instead of `"cd /go/src/github.com/google/cadvisor"`, and add `--rm` so that the container is removed after it's finished;

```Makefile
@docker run -i --rm -v $(PWD):/go/src/github.com/google/cadvisor -w /go/src/github.com/google/cadvisor golang:$(goversion) /bin/sh -c "go get github.com/tools/godep && $(GO) test -tags test -short -race $(pkgs)"
```